build_tools/shared.sh

```diff
@@ -14,3 +14,14 @@ get_dep() {
         echo "$package==$(python sklearn/_min_dependencies.py $package)"
     fi
 }
+
+show_installed_libraries(){
+    # use conda list when inside a conda environment. conda list shows more
+    # info than pip list, e.g. whether OpenBLAS or MKL is installed as well as
+    # the version of OpenBLAS or MKL
+    if [[ -n "$CONDA_DEFAULT_ENV" ]]; then
```

Comment:
For our CI, I think we can use `DISTRIB` as the condition.

```suggestion
    if [[ "$DISTRIB" == *conda* ]] then
```

Reply:
> Maybe you find using CONDA_DEFAULT_ENV too brittle?

The env variable is not well documented, so I was unsure how much we can depend on it. I prefer `CONDA_PREFIX` since it is at least mentioned in the [conda-build docs](https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#environment-variables-that-affect-the-build-process)

I am also okay with checking that the `conda` command exists. How do you think it is brittle?

Reply:
> The env variable is not well documented, so I was unsure how much we can depend on it. I prefer CONDA_PREFIX since it is at least mentioned in the conda-build docs

Makes sense, I switch to using `CONDA_PREFIX`

> I am also okay with checking that the conda command exists. In what situations do you think it is brittle?

Probably a bit edge-casy in the CI context, but on my machine, `conda` always exists as a command and that does not necessarily mean I am in a conda environment. I may have done `conda deactivate` (getting the system python) or I may have activated a venv.